### PR TITLE
fix: Initialize the route only once when using eager server load

### LINKED ISSUE #16046: HasUrlParameter.setParameter is executed twice with vaadin.eagerServerLoad=true
### Description of the bug

I have a View which  `implements HasUrlParameter<String>`. 

Also I configured:

`vaadin.eagerServerLoad=true` (for IndexHtmlRequestListener)

with such setup, `HasUrlParameter.setParameter()` method is executed twice during the first page load

### Expected behavior

`HasUrlParameter.setParameter()` method is executed only one time during the first page load with `vaadin.eagerServerLoad=true`

### Versions

- Vaadin / Flow version: 23.3.6

### LINKED ISSUE #16496: Eager server load does not show the eager version
### Description of the bug

When you enable eager server load using `vaadin.eagerServerLoad=true` then the UI is created during the index page request and the initial UIDL is sent to the browser. In the browser, the UIDL is processed, the state tree is updated and the DOM elements are created. Unfortunately, the DOM elements are not created inside the designated "outlet" div but appended to the body tag. This means they are rendered right below the viewport so nobody can see them. It works however, because after this an xhr is sent to the server (the same as when not using eager mode) and this xhr triggers navigation again, causing #16046 and probably other issues. The result of the second navigation is sent back to the client, and now the elements rendered outside the viewport are removed and the second view instance is rendered correctly.

### Expected behavior

When you enable eager server load, there is only one server request and the view is rendered so you can see it.

### Minimal reproducible example

See `flow-tests/test-eager-bootstrap`

### Versions

- Vaadin / Flow version: main

### PR DESCRIPTION
**Pull request by Artur-**
Changes bootstrap for eager mode so that the index page contains the <flow-container-[appId]> div and the virtual child for this element is created eagerly on the server. This allows the initial render on the client side to put the content inside the flow-container instead of document body as happened before.
    
By using a convention on how the container div is created and id is set, we no longer need to communicate the element tag and id from the client.
    
Fixes #16046
Fixes #16496

### CODE REVIEW

**Comment by Artur-** [comment]
This depends on #16497

File: flow-tests/test-eager-bootstrap/src/test/java/com/vaadin/flow/eagerbootstrap/ParameterIT.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return "/parameter";
+    }
+
+    @Test
+    public void viewInitializedOnce() {
+        openWithParameter("foo");
+        int firstInstance = getInstance();
+        openWithParameter("foo");
+        Assert.assertEquals(firstInstance + 1, getInstance());
+    }
```
Comment (mcollovati) [inline flow-tests/test-eager-bootstrap/src/test/java/com/vaadin/flow/eagerbootstrap/ParameterIT.java:21]: This test passes also without the fix if it runs isolated.
Is there another way to ensure that the view is not initialized twice?

**Review by mcollovati — CHANGES_REQUESTED** [review]

File: flow-tests/test-eager-bootstrap/src/test/java/com/vaadin/flow/eagerbootstrap/ParameterIT.java (hunk 1 above)
Comment (Artur-) [reply flow-tests/test-eager-bootstrap/src/test/java/com/vaadin/flow/eagerbootstrap/ParameterIT.java:21]: Right, the real test is `setParameterCalledAsExpected` and apparently the same view instance is reused even though navigation happens twice. Maybe we can just delete this test as `setParameterCalledAsExpected` ensures that `setParameter` is only called once

File: flow-server/src/main/java/com/vaadin/flow/component/UI.java [hunk 2]
```diff
@@ -1692,21 +1693,17 @@ public void connectClient(String clientElementTag, String clientElementId,
...
                     event -> renderViewForRoute(event.getLocation(),
                             event.getTrigger()));
 
-            // Render the flow view that the user wants to navigate to.
-            renderViewForRoute(location, navigationTrigger);
+            if (!getInternals().getActiveRouterTargetsChain().isEmpty()) {
+                // The view has already been rendered eagerly
+            } else {
+                renderViewForRoute(location, navigationTrigger);
+            }
```
[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/component/UI.java:1706]: Fixed

File: flow-tests/test-eager-bootstrap/src/test/java/com/vaadin/flow/eagerbootstrap/ParameterIT.java (hunk 1 above)
Comment (Artur-) [reply flow-tests/test-eager-bootstrap/src/test/java/com/vaadin/flow/eagerbootstrap/ParameterIT.java]: Removed and simplified

**Review by mcollovati — APPROVED** [review]

**Pull request merged**